database.go

```diff
@@ -149,12 +163,104 @@ func (db *Database) AddBinary(binaryContent []byte) *Binary {
 	return db.Content.Meta.Binaries.Add(binaryContent, WithKDBXv31Binary)
 }
 
+type BinarySet map[*Binary]bool
+type BinaryDuplicates map[*Binary]*Binary
+type BinaryUsages map[int][]BinaryReference
+
+func (db *Database) CleanupBinaries() int {
+	binaries := Binaries{}
+	usages := db.getBinariesMap()
+	garbage := db.FindBinaryGarbage(usages)
+	duplicates := db.FindBinaryDuplicates()
+
+	for _, binary := range *db.GetBinaries() {
+		if _, ok := garbage[&binary]; ok {
```

Comment:
Is there any particular reason to work with pointer here instead of with the ID?

Reply:
No, just used BinarySet type as common approach.

Reply:
Actually, I think, yes. Since I change IDs in this loop.

Reply:
Yeah, but you do not need to change IDs for the garbage.

There it would be sufficient to find it by ID.

Reply:
Yes, already replaced it with, we even don't need to call find because we are already in binaries iteration loop.